hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/ProcedureExecutor.java

```diff
@@ -2070,6 +2070,9 @@ private void execCompletionCleanup(Procedure<TEnvironment> proc) {
       // Catch NullPointerExceptions or similar errors...
       LOG.error("CODE-BUG: uncatched runtime exception for procedure: " + proc, e);
     }
+    // call schedulers completion cleanup, we have some special clean up logic in this method so if
+    // it throws any exceptions, we can not just ignore it, like the above procedure's cleanup
+    scheduler.completionCleanup(proc);
   }
 
   private void procedureFinished(Procedure<TEnvironment> proc) {
```

Comment:
Can you rename this method or add some javadoc that says it is intended to be called only for root procedure cleanup? I can see this is the case when I trace the code, but it's not at all clear on unassisted code review.